### PR TITLE
Add metomi-isodatetime recipe for conda-forge

### PR DESCRIPTION
**Pull request by kinow**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by kinow** [comment]
>License file is packaged (see here for an example)

Wasn't sure if it would pick up the file from the PYPI archive, so copied the `LICENSE` file from the original source code in the new recipe folder.

>A tarball (url) rather than a repo (e.g. git_url) is used in your recipe (see

Ops, accidentally used the git revision. Push-forcing a commit fixing it, to use the package from PYPI. 

>GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

I've added only myself as maintainer for now (@hjoliver / @matthewrmshin let me know who else should be added here as maintainer for metomi-isodatetime)

**Comment by kinow** [comment]
I wasn't going to work on this now, as I thought it was complicated. I had read some documentation about it, but then I saw [this tweet](https://twitter.com/xhochy/status/1173182116689059840) from @xhochy explaining to somebody else how easy it was.

@xhochy you were right, it isn't that hard. We used this recipe in [a private channel](https://github.com/cylc/cylc-conda/tree/0b93dc80f161318e179673ca9da54ca852dafec7/recipes/metomi-isodatetime) already, but there could be something missing in order to publish it to Conda Forge. Not in a hurry to get it review/merged :+1: so happy to ping the Python team or ask someone on gitter to have a look in a few days/weeks.

Thanks!
Bruno

File: recipes/metomi-isodatetime/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: a8dd8a12f4df79098c297de192d6dd75b93c78a2565f10ec13281f2f2c5faf29
+
+build:
+  skip: True  # [not linux]
```
Comment (xhochy) [inline recipes/metomi-isodatetime/meta.yaml]: This looks like a general Python package without any binary code, so the conda package is neither specific to the OS nor the Python version and can therefore be `noarch: python`
```suggestion
  noarch: python
```

Comment (kinow) [reply recipes/metomi-isodatetime/meta.yaml]: I saw this in the example recipe, but wasn't sure if this would work. Thanks @xhochy !

**Comment by kinow** [comment]
That was really quick and easy! Thanks a lot @xhochy !!!

**Comment by xhochy** [comment]
Also sent you two upstream PRs to make the recipe simpler on the next release of `isodatetime`.

**Pull request merged**